bayes/jacobian.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+
+
+def d_model_error_d_vector(model_error, number_vector):
+    """
+    Calculates the derivative of `model_error` w.r.t `number_vector`
+
```

Comment:
In the variational bayes algorithm when the user only provides the `__call__` but not `jacobian`, e.g. in "tests/test_vb.py"

Reply:
I personally feel that we should not mix up the interfaces (parameter vector vs named dict). I think this becomes confusing at some point and is the core feature of this interface here. We are providing a clearer interfaces with named parameters to complex data situations. In case you only have single vector/parameter, just use the optimization/inference procedure that takes vectors and matrices directly.